node/src/crypto/asymmetric_key.rs

```diff
@@ -37,8 +36,16 @@ pub fn sign<T: AsRef<[u8]>>(
             Signature::Ed25519(signature.to_bytes())
         }
         (SecretKey::Secp256k1(secret_key), PublicKey::Secp256k1(_public_key)) => {
-            let signer = ecdsa::Signer::new(secret_key).expect("should create secp256k1 signer");
-            Signature::Secp256k1(signer.sign_with_rng(rng, message.as_ref()))
+            let secret_key =
+                secp256k1::SecretKey::parse(secret_key).expect("should create secp256k1 secret");
```

Comment:
(As discussed, it's because it's not `Copy`.)

Reply:
But it's `Clone`, right? I think we should not be parsing the secret for every `sign` operation – it makes it looks like it can pass on the try number 1321 and fail on 1322.

Reply:
While I agree that is the clean way to do this, `secp256k1::SecretKey` and friends don't implement `Copy` and `AsRef`